scene/resources/skeleton_modification_2d.h

```diff
@@ -0,0 +1,620 @@
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef SKELETONMODIFICATION2D_H
+#define SKELETONMODIFICATION2D_H
+
+#include "scene/2d/skeleton_2d.h"
+
+///////////////////////////////////////
+// SkeletonModificationStack2D
+///////////////////////////////////////
+
+class Skeleton2D;
+class SkeletonModification2D;
+class Bone2D;
+
+class SkeletonModificationStack2D : public Resource {
+	GDCLASS(SkeletonModificationStack2D, Resource);
+	friend class Skeleton2D;
+	friend class SkeletonModification2D;
+
+protected:
+	static void _bind_methods();
+	void _get_property_list(List<PropertyInfo> *p_list) const;
+	bool _set(const StringName &p_path, const Variant &p_value);
+	bool _get(const StringName &p_path, Variant &r_ret) const;
+
+public:
+	Skeleton2D *skeleton;
+	bool is_setup = false;
+	bool enabled = true;
+	float strength = 0.0;
+
+	enum EXECUTION_MODE {
+		execution_mode_process,
+		execution_mode_physics_process
+	};
+
+	Vector<Ref<SkeletonModification2D>> modifications;
+	int modifications_count = 0;
+
+	void setup();
+	void execute(float delta, int p_execution_mode);
+
+	bool editor_gizmo_dirty = false;
+	void draw_editor_gizmos();
+	void set_editor_gizmos_dirty(bool p_dirty);
+
+	void enable_all_modifications(bool p_enable);
+	Ref<SkeletonModification2D> get_modification(int p_mod_idx) const;
+	void add_modification(Ref<SkeletonModification2D> p_mod);
+	void delete_modification(int p_mod_idx);
+	void set_modification(int p_mod_idx, Ref<SkeletonModification2D> p_mod);
+
+	void set_modification_count(int p_count);
+	int get_modification_count() const;
+
+	void set_skeleton(Skeleton2D *p_skeleton);
+	Skeleton2D *get_skeleton() const;
+
+	bool get_is_setup() const;
+
+	void set_enabled(bool p_enabled);
+	bool get_enabled() const;
+
+	void set_strength(float p_strength);
+	float get_strength() const;
+
+	SkeletonModificationStack2D();
+};
+
+///////////////////////////////////////
+// SkeletonModification2D
+///////////////////////////////////////
+
+class SkeletonModification2D : public Resource {
+	GDCLASS(SkeletonModification2D, Resource);
+	friend class Skeleton2D;
+	friend class Bone2D;
+
+protected:
+	static void _bind_methods();
+
+	SkeletonModificationStack2D *stack;
+	int execution_mode = SkeletonModificationStack2D::EXECUTION_MODE::execution_mode_process;
+
+	bool enabled = true;
+	bool is_setup = false;
+	bool execution_error_found = false;
+
+	bool _print_execution_error(bool p_condition, String p_message);
```

Comment:
Please, consider remove this function in favour of `ERR_FAIL_COND....`.

Reply:
The reason this function was added is because the problem before was it would spam the console with errors when its not setup correctly, something that I changed based on user feedback and testing. I can instead use `ERR_FAIL_COND...` and similar, like I was before, but then the issue of error log spamming may occur again.

Reply:
Use ERR_PRINT_ONCE